mopidy_tidal/library.py

```diff
@@ -505,9 +496,10 @@ def _get_artist_top_tracks(session, artist_id):
 
     def _lookup_artist(self, session, parts):
         artist_id = parts[2]
-        artist_uri = ':'.join(parts)
+        artist_uri = ":".join(parts)
 
         tracks = self._artist_cache.get(artist_uri)
+        # NOTE: unreachable?
```

Comment:
We should find out if these unreachable parts are indeed unreachable, when they were introduced and why they were needed in the first place.

Reply:
I think there's some duplication in terms of checking for the cache both in the calling fn and in the getters.  Proably it would make sense to move it all to the getters, but I've not thought very hard about it.  I think these lines are unreachable because the calling fn also checks for a cache hit.  I've not checked whether we might somehow populate the cache with a sub type (e.g. track when getting an album?)

Reply:
Some were added quite recently, eg. 831ebc09edb924c3aa98ddd35aa30093c7d503f9.  @BlackLight I'm sure I'm missing something here!

Reply:
@2e0byo I think you're right on this one, that code is most likely unreachable. It returns only the playlist tracks, not a tuple with both the tracks and the playlist object. The consumer, however, expects both of them to be there. So if the code hasn't broken so far, it's most likely because that branch can't be reached. That's because I had moved the cache check in the upstream caller, but I forgot to remove it from here.

Reply:
@BlackLight that was my guess, that makes sense.  I'll remove these checks then. :D